algorithms/src/snark/varuna/ahp/indexer/circuit.rs

```diff
@@ -123,14 +110,19 @@ impl<F: PrimeField, MM: SNARKMode> Circuit<F, MM> {
         self.index_info.max_degree::<F, MM>()
     }
 
-    /// The size of the constraint domain in this R1CS instance.
-    pub fn constraint_domain_size(&self) -> usize {
-        crate::fft::EvaluationDomain::<F>::new(self.index_info.num_constraints).unwrap().size()
-    }
-
-    /// The size of the variable domain in this R1CS instance.
-    pub fn variable_domain_size(&self) -> usize {
-        crate::fft::EvaluationDomain::<F>::new(self.index_info.num_variables).unwrap().size()
+    /// The maximum domain size.
+    pub fn max_domain_size(&self) -> usize {
+        let size = [
+            2 * self.index_info.num_constraints, // zerocheck poly degree
+            2 * self.index_info.num_variables,   // lineval sumcheck poly degree
+            2 * self.index_info.num_non_zero_a,  // matrix sumcheck poly degree
+            2 * self.index_info.num_non_zero_b,  // matrix sumcheck poly degree
+            2 * self.index_info.num_non_zero_c,  // matrix sumcheck poly degree
+        ]
+        .into_iter()
+        .max()
+        .unwrap();
+        crate::fft::EvaluationDomain::<F>::compute_size_of_domain(size).unwrap()
```

Comment:
Should this also live with `CircuitInfo`, just like `get_degree_bounds`?

Reply:
(In general there are bunch of such utility functions which are kind of distributed between `AHPForR1CS`, `Circuit`, and `CircuitInfo`; these should be audited and we should clarify why each is in its current place (or move it if appropriate))

Reply:
Very much agree, I left `max_degree(...)` within ahp.rs because it is also used in other tests to directly infer the size needed for the SRS.

[3383c94](https://github.com/AleoHQ/snarkVM/pull/1955/commits/3383c94abdbb491b2e9ab6f5874172b8f4b43bf7)